src/components/SearchView.tsx

```diff
@@ -139,32 +137,40 @@ export class SearchView extends React.Component<SearchViewProps, SearchViewState
     }
 
     onCategoriesChanged(categories: string[]) {
-        this.setState({ selectedCategories: categories })
+        this.setState({ selectedCategories: categories });
     }
 
-    updateSearchView(text: string) {
-        if (this.state.displayMode === "list") {
+    updateSearchViewDelayed(text: string) {
+        if (text.length > 0 && !this.state.structured) {
             this.props.libraryContainer.raiseEvent("searchTextUpdated", text);
         }
 
         this.setState({ searchText: text });
-
-        // Update library container of current search
         this.props.onSearchModeChanged(text.length > 0);
     }
 
+    clearSearch() {
+        let searchInput: any = document.getElementById("SearchInputText");
```

Comment:
I'm not sure if `getElementById` or `ref` is the best way to identify a component, but let's leave this as-is.